src/Processors/Transforms/FilterSortedStreamByRange.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <Interpreters/ExpressionActions.h>
+#include <Processors/ISimpleTransform.h>
+#include <Processors/Transforms/ExpressionTransform.h>
+#include <Processors/Transforms/FilterTransform.h>
+
+namespace DB
+{
+
+/// Could be used when the predicate given by expression_ is true only on one continuous range of values and input is monotonous by that value.
```

Comment:
I rephrased the comment slightly, pls, check if it is better now.
the logic here is a bit different. we filter by predicate "does this row contains PK value that belongs to some semi-interval `[B; E)` of PK values". since data comes in order, in general this predicate will be 0 for rows before B, then it will change its value to 1 and then it again will start to have value 0 for rows since E.

Reply:
The new version is okay, tnx.